comparison.py

```diff
@@ -1,29 +1,64 @@
 import numpy as np
 from hmc_sampler import LeapfrogHMC
 from hmc_sampler import U7HMC
+import gelman_rubin
 
+
+number_of_chains = 2
 chain_length = 5000
-#Länge der Schrittweite im Leapfrog
 stepsize = 1.5
-#Anzahl der Schritte im Leapfrog
 trajectory_length = 10
-
 dimension = 2
 
-x_0 = np.zeros(dimension)
+x_0 = np.zeros(dimension)   #may be chosen randomly for different chains
+
+
 def log_prob(x):
-    return -0.5* np.sum(x**2 )
-#gradient der log Dichte funktion
+
+    return -1./2 * np.sum(x**2)
```

Comment:
```suggestion
    return -0.5 * np.sum(x**2)
```
I usually try to keep formulas in code as easy to read as possible. This and the following two suggestions are what I would write, but it's probably quite subjective.